src/EventGridExtension/EventGridListener.cs

```diff
@@ -16,28 +16,27 @@ public EventGridListener(ITriggeredFunctionExecutor executor, EventGridExtension
             _listenersStore = listenersStore;
             _functionName = functionName;
             Executor = executor;
+
+            // Register the listener as part of create time initialization
+            _listenersStore.AddListener(_functionName, this);
         }
```

Comment:
Does this still return 404 if the function is disabled? I'm pretty sure that isn't based on the listener getting started, but wanted to double-check. I was brainstorming what could be affected by this change and that was about all I could think of...